woodwork/table_accessor.py

```diff
@@ -1167,14 +1167,14 @@ def _check_index(dataframe, index):
         raise ColumnNotPresentError(
             f"Specified index column `{index}` not found in dataframe"
         )
-    if (
-        index is not None
-        and isinstance(dataframe, pd.DataFrame)
-        and not dataframe[index].is_unique
-    ):
-        # User specifies an index that is in the dataframe but not unique
-        # Does not check for Dask as Dask does not support is_unique
-        raise IndexError("Index column must be unique")
+    if index is not None and isinstance(dataframe, pd.DataFrame):
+        if not dataframe[index].is_unique:
+            # User specifies an index that is in the dataframe but not unique
+            # Does not check for Dask as Dask does not support is_unique
```

Comment:
Minor detail, but I think we should update this comment a bit. We are now skipping both the unique check and the null check for Dask and Koalas, with the primary reason being that we don't want to pull the data into memory to perform the check on initialization. Maybe move it above the `if not dataframe[index].is_unique:` line and update the text to better match the new behavior?

Reply:
updated